sympy/printing/str.py

```diff
@@ -814,21 +818,44 @@ def _print_set(self, s):
             return "set()"
         return '{%s}' % args
 
-    def _print_FiniteSet(self, s):
+    def _print_Set(self,s):
+        from sympy.sets.sets import FiniteSet, Interval, ProductSet, \
+            Complement
+        from sympy.sets.conditionset import ConditionSet
+        from sympy.sets.fancysets import Range, ImageSet
+        from sympy.geometry.entity import GeometrySet
+        if isinstance(s, Interval):
+            return self._print_Interval(s)
+        if isinstance(s, ConditionSet):
+            return self._print_ConditionSet(s)
+        name = s.__class__.__name__
+        items = None
+        if isinstance(s, (ProductSet, Complement, Range, ImageSet,
+                          GeometrySet)):
+            items = s.args
+        else:
+            items = sorted(s.args, key=default_sort_key)
+        args = ''
+        for item in items:
+            if isinstance(item, FiniteSet):
+                args += self._wrapped_Finite_Set_print(item) + ', '
+            else:
+                args += self._print(item) + ', '
+        args = args[:-2]
+        return name + '({})'.format(args)
```

Comment:
I'm a bit skeptical about if the implementation needs to be complicated like this though.
Why can't `_print_FiniteSet` be overridden?

Reply:
I tried just overriding `_print_FiniteSet` but that did not work because of the way things cascade through the printing of sets. Most of this is to make sure that a `FiniteSet` wrapped in something else prints out as just `{...}`.